data/changelog/posts/ocaml-lsp/2025-08-20-ocaml-eglot-brings-lsp-support-to-emacs.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Emacs Integration for OCaml LSP Server: Introducing ocaml-eglot
+tags: [ocaml-lsp, platform]
+---
+
+# Emacs Integration for OCaml LSP Server: Introducing ocaml-eglot
+
+## TL;DR
+
+`ocaml-eglot` provides full OCaml language support in Emacs through the Language Server Protocol (LSP) instead of direct Merlin integration. It offers the same features as `merlin.el` with simplified setup and enhanced capabilities like project-wide search. If you're starting fresh or want a more standardized approach, try `ocaml-eglot`. If your current `merlin.el` setup works well, you can continue using it.
+
+**Quick start**: Install `ocaml-lsp-server`, add `ocaml-eglot` to your Emacs config, get the same OCaml development experience with less configuration.
+
+## What is ocaml-eglot?
+
+`ocaml-eglot` connects Emacs to `ocaml-lsp-server` using the Language Server Protocol, providing a standardized way to get OCaml language support. This eliminates editor-specific integrations while preserving all OCaml-specific features.
```

Comment:
It is worth mentioning that since the recent versions of Emacs (`29`), `Eglot`, an `lsp` client, has been shipped with Emacs. It is therefore reasonable to use `ocaml-lsp-server` to provide IDE services (standardising the experience for users of multiple languages). However, LSP is designed to be generic, so relying solely on the features of LSP and Eglot limits functionality and offers fewer features than `merlin.el`. That is why the server has been extended to support more features, and `ocaml-eglot` allows you to benefit from these features in Emacs, in conjunction with `eglot`.